File: example/depth/sandbox/2-test/bar/test/src/bar/BarTests.java
package bar;

import static org.junit.Assert.assertEquals;

import java.nio.file.*;
import org.junit.Test;

public class BarTests {
  @Test
  public void simple() throws Exception {
    String result = Bar.generateHtml("world");
    Path path = Paths.get("generated.html");
    Files.write(path, result.getBytes());
    assertEquals("<p>world</p>", Files.readString(path));
  }
}

File: example/scalalib/web/8-cross-platform-version-publishing/bar/test/src/BarTests.scala
package bar
import utest._
object BarTests extends TestSuite {
  def tests = Tests {
    test("test") {
      val result = Bar.value.toString
      val matcher = "<p>world Specific code for Scala [23].x</p>".r
      assert(matcher.matches(result))
      result
    }
  }
}
